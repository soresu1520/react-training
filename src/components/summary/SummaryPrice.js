import styled from "styled-components";

const SummaryPrice = ({ price, deliveryPrice }) => {
  return (
    <PriceDiv>
      <SubPriceDiv>
        <PriceText>Subtotal</PriceText>
        <PriceTextRight>{price} $</PriceTextRight>
      </SubPriceDiv>
      <SubPriceDiv>
        <PriceText>Delivery</PriceText>
        <PriceTextRight>{deliveryPrice.toFixed(2)} $</PriceTextRight>
      </SubPriceDiv>
      <Line></Line>
      <SubPriceDiv>
        <TotalText>Total</TotalText>
        <TotalTextRight data-testid="totalPrice">
          {(+price + deliveryPrice).toFixed(2)} $
        </TotalTextRight>
      </SubPriceDiv>
    </PriceDiv>
  );
};

export default SummaryPrice;

const PriceDiv = styled.div`
  background: var(--color-greyscale-50);
  padding: 1.5em;
  display: flex;
  flex-flow: column nowrap;
  flex-basis: 50%;
`;

const SubPriceDiv = styled.div`
  display: flex;
  flex-flow: row nowrap;

  @media (max-width: 550px) {
    flex-flow: column nowrap;
  }
`;

const Line = styled.div`
  border-bottom: 1px solid var(--color-dark-icon);
`;

const PriceText = styled.h4`
  font-size: var(--font-size-h4);
  font-weight: var(--font-weight-heading);
  color: var(--color-greyscale-600);
  margin-top: 0;
  margin-bottom: 0.5em;
  flex-basis: 50%;

  @media (max-width: 550px) {
    text-align: center;
  }
`;

const PriceTextRight = styled(PriceText)`
  text-align: right;

  @media (max-width: 550px) {
    text-align: center;
  }
`;

const TotalText = styled.h2`
  color: var(--color-brand-500);
  font-size: var(--font-size-h2);
  font-weight: var(--font-weight-heading);
  margin-top: 0.5em;
  margin-bottom: 0;
  flex-basis: 40%;

  @media (max-width: 550px) {
    text-align: center;
  }
`;

const TotalTextRight = styled(TotalText)`
  text-align: right;
  flex-basis: 60%;

  @media (max-width: 550px) {
    text-align: center;
  }
`;
