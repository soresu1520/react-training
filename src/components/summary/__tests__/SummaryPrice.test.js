import renderer from "react-test-renderer";
import { render } from "@testing-library/react";
import SummaryPrice from "../SummaryPrice";

const price = 7;
const deliveryPrice = 2;

it("summary price renders correctly", () => {
  const tree = renderer
    .create(<SummaryPrice price={price} deliveryPrice={deliveryPrice} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("total price is a sum of price and delivery price", async () => {
  const summaryPrice = render(<SummaryPrice price={price} deliveryPrice={deliveryPrice} />);
  const totalPrice = await summaryPrice.findByTestId("totalPrice");
  expect(totalPrice).toHaveTextContent(`${(price + deliveryPrice).toFixed(2)} $`);
});
