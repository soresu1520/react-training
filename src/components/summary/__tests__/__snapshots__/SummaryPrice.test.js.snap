// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`summary price renders correctly 1`] = `
<div
  className="sc-bczRLJ guFtRN"
>
  <div
    className="sc-gsnTZi inqXkk"
  >
    <h4
      className="sc-hKMtZM cwedSL"
    >
      Subtotal
    </h4>
    <h4
      className="sc-hKMtZM sc-eCYdqJ cwedSL kEfZbk"
    >
      7
       $
    </h4>
  </div>
  <div
    className="sc-gsnTZi inqXkk"
  >
    <h4
      className="sc-hKMtZM cwedSL"
    >
      Delivery
    </h4>
    <h4
      className="sc-hKMtZM sc-eCYdqJ cwedSL kEfZbk"
    >
      2.00
       $
    </h4>
  </div>
  <div
    className="sc-dkzDqf gBJxRS"
  />
  <div
    className="sc-gsnTZi inqXkk"
  >
    <h2
      className="sc-jSMfEi fpfkPe"
    >
      Total
    </h2>
    <h2
      className="sc-jSMfEi sc-gKXOVf fpfkPe bmARjz"
      data-testid="totalPrice"
    >
      9.00
       $
    </h2>
  </div>
</div>
`;
